plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2634,10 +2634,14 @@ private static void updatePartitioning(Table icebergTable, Transaction transacti
         else {
             PartitionSpec partitionSpec = parsePartitionFields(schema, partitionColumns);
             Set<PartitionField> partitionFields = ImmutableSet.copyOf(partitionSpec.fields());
-            difference(existingPartitionFields, partitionFields).stream()
+
+            existingPartitionFields.stream()
+                    .filter(existing -> partitionFields.stream().noneMatch(newField -> newField.equals(existing)))
                     .map(PartitionField::name)
                     .forEach(updatePartitionSpec::removeField);
-            difference(partitionFields, existingPartitionFields)
+
+            partitionFields.stream()
```

Comment:
I think it would be helpful to leave a comment explaining why we can’t use `difference` or other collection operations here